exchanges/bybit/convert.go

```diff
@@ -0,0 +1,280 @@
+	ToAmount     types.Number    `json:"toAmount"`
+	ExpiredTime  types.Time      `json:"expiredTime"` // The expiry time for this quote (15 seconds)
+	RequestID    string          `json:"requestId"`
+	ExtTaxAndFee json.RawMessage `json:"extTaxAndFee"` // Compliance-related field. Currently returns an empty array, which may be used in the future
+}
+
+// RequestAQuote requests a conversion quote between two coins with the specified parameters.
+func (e *Exchange) RequestAQuote(ctx context.Context, params *RequestAQuoteParams) (*RequestAQuoteResponse, error) {
+	if !slices.Contains(supportedAccountTypes, params.AccountType) {
+		return nil, fmt.Errorf("%w: %q", errUnsupportedAccountType, params.AccountType)
+	}
+
+	if params.From.IsEmpty() {
+		return nil, fmt.Errorf("from %w", currency.ErrCurrencyCodeEmpty)
+	}
+	if params.To.IsEmpty() {
+		return nil, fmt.Errorf("to %w", currency.ErrCurrencyCodeEmpty)
+	}
+
+	if params.From.Equal(params.To) {
+		return nil, errCurrencyCodesEqual
+	}
+
+	if !params.RequestCoin.IsEmpty() {
+		if !params.RequestCoin.Equal(params.From) {
+			return nil, errRequestCoinInvalid
+		}
+	} else {
+		params.RequestCoin = params.From
+	}
+
+	if params.Amount <= 0 {
+		return nil, fmt.Errorf("amount %w", order.ErrAmountIsInvalid)
+	}
+
+	payload := &struct {
+		AccountType  string `json:"accountType"`
+		From         string `json:"fromCoin"`
+		To           string `json:"toCoin"`
+		Amount       string `json:"requestAmount"`
+		RequestCoin  string `json:"requestCoin"`
+		FromCoinType string `json:"fromCoinType,omitempty"`
+		ToCoinType   string `json:"toCoinType,omitempty"`
+		ParamType    string `json:"paramType,omitempty"`
+		ParamValue   string `json:"paramValue,omitempty"`
+		RequestID    string `json:"requestId,omitempty"`
+	}{
+		AccountType:  string(params.AccountType),
+		From:         params.From.Upper().String(),
+		To:           params.To.Upper().String(),
+		Amount:       strconv.FormatFloat(params.Amount, 'f', -1, 64),
+		RequestCoin:  params.RequestCoin.Upper().String(),
+		FromCoinType: params.FromCoinType,
+		ToCoinType:   params.ToCoinType,
+		ParamType:    params.ParamType,
+		ParamValue:   params.ParamValue,
+		RequestID:    params.RequestID,
+	}
+
+	var resp *RequestAQuoteResponse
+	return resp, e.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodPost, "/v5/asset/exchange/quote-apply", nil, payload, &resp, defaultEPL)
+}
+
+// ConfirmAQuoteResponse represents a response for confirming a quote
+type ConfirmAQuoteResponse struct {
+	ExchangeStatus string `json:"exchangeStatus"`
+	QuoteTxID      string `json:"quoteTxId"`
+}
+
+// ConfirmAQuote confirms a quote transaction and executes the conversion
+func (e *Exchange) ConfirmAQuote(ctx context.Context, quoteTxID string) (*ConfirmAQuoteResponse, error) {
+	if quoteTxID == "" {
+		return nil, errQuoteTxIDEmpty
+	}
+
+	payload := struct {
+		QuoteTxID string `json:"quoteTxId"`
+	}{
+		QuoteTxID: quoteTxID,
+	}
+
```

Comment:
Lot's of whitespace in these funcs. You can keep it, but API implementations are pretty simple and so I don't think it needs it